proto/workload.proto

```diff
@@ -74,8 +74,22 @@ message Service {
   repeated string subject_alt_names = 6;
 }
 
+// Workload represents a workload - an endpoint (or collection behind a hostname).
+// The xds primary key is "uid" as defined on the workload below.
+// Secondary (alias) keys are the unique `network/IP` pairs that the workload can be reached at.
 message Workload {
   // UID represents a globally unique opaque identifier for this workload.
+  // For k8s resources, it is recommended to use the more readable format:
+  //
+  // cluster/group/version/kind/namespace/name/optional-index
+  //
+  // As an example, a ServiceEntry with two WorkloadEntries inlined could become
+  // two Workloads with the following UIDs:
+  // - cluster1/networking.istio.io/v1alpha3/ServiceEntry/default/external-svc/endpoint1
+  // - cluster1/networking.istio.io/v1alpha3/ServiceEntry/default/external-svc/endpoint2
+  //
+  // For VMs and other workloads other formats are also supported; for example,
+  // a single UID string: "0ae5c03d-5fb3-4eb9-9de8-2bd4b51606ba"
   string uid = 20;
```

Comment:
UID not being a uuid feels a bit funky. I get it technical makes sense, just the 2 are typically used interchange-ably. Not saying we should change it necessarily

Reply:
I guess maybe just `id` would make it feel 'right' to me

Reply:
I think uid is slightly better, albeit awkward, because it does _need_ to be universally unique